src/Grids/vertical_coordinates.jl

```diff
@@ -29,12 +29,41 @@ struct StaticVerticalCoordinate{C, D, E, F} <: AbstractVerticalCoordinate
     Δᵃᵃᶜ :: F
 end
 
+struct ZStarVerticalCoordinate{C, D, E, F, H, CC, FC, CF, FF} <: AbstractVerticalCoordinate
+    cᵃᵃᶠ :: C
+    cᵃᵃᶜ :: D
+    Δᵃᵃᶠ :: E
+    Δᵃᵃᶜ :: F
+      ηⁿ :: H
+    σᶜᶜⁿ :: CC
+    σᶠᶜⁿ :: FC
+    σᶜᶠⁿ :: CF
+    σᶠᶠⁿ :: FF
+    σᶜᶜ⁻ :: CC
+    ∂t_σ :: CC
+end
```

Comment:
Isn't this more general than z*? It's just a vertical coordinate that _can_ evolve in time